### PR TITLE
Release machinelearningservices microsoft.machine learning services 2024 07 01 preview

### PR DESCRIPTION
**Pull request by ragovada**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by TimLovellSmith — APPROVED** [review]

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in #24249

**Comment by theodoreschen** [comment]
GuidUsage Swagger LintDiff error was suppressed in PR #29657

**Comment by razvanbadea-msft** [comment]
>         "enableOSPatching": {

in place of having this boolean flags you can convert them in an enum like for example EnabledFeatures with values like [OsPatching, RootAccess, SSO, etc] so that if in a new version some other flags are needed you can extend the enum with new values in a non-breaking way

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/machineLearningServices.json:2167 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>         "scheduledRebootTime": {

what format does this property uses ?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/machineLearningServices.json:2998 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "description": "Describes the jupyter kernel settings for the image if its a custom environment",

small nit: typo

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/machineLearningServices.json:3061 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>         "reference": {

you can add the uri format

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/machineLearningServices.json:3084 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>         "language": {

couldnt this be an enum? how you make sure the values are valid

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/machineLearningServices.json:3279 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "200": {

this is a lro. why are you using also 200?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:189 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>     "ProvisioningState": {

this should be readonly

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:5807 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>         "x-ms-long-running-operation": true

add a location header

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:326 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>               "$ref": "#/definitions/Workspace"

missing provisioningState

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:367 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "200": {

the put needs to have 201 for when creating a resource

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:364 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>               "$ref": "#/definitions/WorkspaceConnectionPropertiesV2BasicResource"

you should use a schema with all properties patchable. this model has required properties

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:597 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "200": {

missing 201 for  resource creation


---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:651 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>               "$ref": "#/definitions/EndpointResourcePropertiesBasicResource"

missing provisioningState

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:2230 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "200": {

same here about 201 for creation resources

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:2227 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "200": {

what this code is used for?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:3250 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by razvanbadea-msft** [comment]
>           "additionalProperties": {}

How would a user of this API know what properties and values might be present and how to consume them?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:4289 in c61c6ff. [](commit_id = c61c6ff2b0c80a47d03fcc124005200041d29ecd, deletion_comment = False)

**Comment by ZhidaLiu** [comment]
> > {
> 
> missing securityDefinitions section
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:1 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

it's on line 7281

**Comment by ZhidaLiu** [comment]
> > ```
> >       "200": {
> > ```
> 
> this is a lro. why are you using also 200?
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:189 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

This is a behavior that always exists in this API, I think we just keep it there to avoid the breaking change.

**Comment by ZhidaLiu** [comment]
> > ```
> >           "$ref": "#/definitions/Workspace"
> > ```
> 
> missing provisioningState
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:367 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

defined in line 4928

**Comment by ZhidaLiu** [comment]
> > ```
> >       "200": {
> > ```
> 
> the put needs to have 201 for when creating a resource
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:364 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

This is a very old API when there is no 201 requirement, we return 202 instead of 201 for the LRO creation response and same for a lot of other APIs

**Comment by ZhidaLiu** [comment]
> > ```
> >           "$ref": "#/definitions/WorkspaceConnectionPropertiesV2BasicResource"
> > ```
> 
> you should use a schema with all properties patchable. this model has required properties
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:597 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

Good suggestion, but we might not have time to fix in this version, as this API definition has always been like this we would like to fix this in the next API version.

**Comment by ZhidaLiu** [comment]
> > ```
> >           "$ref": "#/definitions/EndpointResourcePropertiesBasicResource"
> > ```
> 
> missing provisioningState
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:2230 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

This is a proxy resource that only used by some of our partner services right now, we are on our way to retiring this resource since we plan to implement it in a different way, so might now worth to fully supporting the ProvisioningState.

**Comment by ZhidaLiu** [comment]
> > ```
> >       "200": {
> > ```
> 
> what this code is used for?
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:3250 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

I think we actually do not return 200 for delete requests anymore, but would like to remove this in the next version as this has been introduced in the previous version already

**Comment by ZhidaLiu** [comment]
> > ```
> >       "additionalProperties": {}
> > ```
> 
> How would a user of this API know what properties and values might be present and how to consume them?
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json:4289 in [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd). [](commit_id = [c61c6ff](https://github.com/Azure/azure-rest-api-specs/commit/c61c6ff2b0c80a47d03fcc124005200041d29ecd), deletion_comment = False)

This a diagnose result of the resource's setup, the customer will call this API to just view the result, they will need to do any operation on the result except read it.

**Comment by ragovada** [comment]
Avoid additional properties error for serverless endpoint inference headers was already supressed and approved in last GA https://github.com/Azure/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/readme.md

**Comment by ZhidaLiu** [comment]
I don't think there are new changes introduce for file machinelearningService.json, so it's the same as the last preview API version, we will have a thread with the owner on your feedbacks.

**Comment by ragovada** [comment]
FYI ARM Reviewer- Manifest release for this API version has already been completed and the API has already been released globally. We want to release the specs now as soon as possible.

**Comment by razvanbadea-msft** [comment]
> FYI ARM Reviewer- Manifest release for this API version has already been completed and the API has already been released globally. We want to release the specs now as soon as possible.

added the signed off label based on this. Having this kind of exceptions (adding LintDiff: 328 Errors, 1501 Warnings just on this pr) makes the process of reviewing the pr very hard to follow, you should take this into considerations for future modifications so that the pr will be reviewed faster.
